File: 287-FindTheDuplicateNumber/287-FindTheDuplicateNumber.java
// Last updated: 2/27/2026, 5:19:24 PM
class Solution {
    public int findDuplicate(int[] nums) {
        
        int idx;
        for(int i = 0; i < nums.length; i++){

            idx = Math.abs(nums[i]);

            if(nums[idx] < 0) return idx;

            nums[idx] = -nums[idx];

        }

        return 0;


    }
}

File: 287-FindTheDuplicateNumber/287-FindTheDuplicateNumber.py
# Last updated: 7/12/2026, 6:20:29 PM
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:

        for i in range(len(nums)):
            idx = abs(nums[i])
            if nums[idx] < 0:
                return idx

            nums[idx] = -nums[idx] 

        return 0
        